### PR TITLE
refactor: perfom linting for punkt.py

### PR DESCRIPTION
**Pull request by 12mohaned**
This pull request refactors the punks.py file that contain PunktTokenizer class by doing the following:- 

- Removing un-used variables. 
- Renaming variables to follow PEP8 standard.
- Removing dead code or un-used keywords and variables. 
- Removing un-necessary else and if's. 
- Replace '==' in checking boolean values with the variable value. For example refactor if value == False with if not value: (performance reasons).

### CODE REVIEW

**Review by tomaarsen — COMMENTED** [review]
I like the look of this PR! I do have some comments though. Also, it seems that there was a small error in the PR somewhere causing the CI to fail. You can check the outputs for debugging. Perhaps you've renamed a variable to something that was already being used for an existing variable?

File: nltk/tokenize/punkt.py [hunk 1]
```diff
@@ -1583,13 +1581,10 @@ def _second_pass_annotation(self, aug_tok1, aug_tok2):
...
 
-            if is_sent_starter == False:
+            if not is_sent_starter:
                 aug_tok1.sentbreak = False
                 aug_tok1.abbr = True
-                if tok_is_initial:
-                    return REASON_INITIAL_WITH_ORTHOGRAPHIC_HEURISTIC
-                else:
-                    return REASON_NUMBER_WITH_ORTHOGRAPHIC_HEURISTIC
+                return REASON_INITIAL_WITH_ORTHOGRAPHIC_HEURISTIC
```
Comment (tomaarsen) [inline nltk/tokenize/punkt.py]: This seems off - why is the option for `REASON_NUMBER_WITH_ORTHOGRAPHIC_HEURISTIC` removed? is `tok_is_initial` always True? I wouldn't blame you for thinking that both branches of the if-else were identical, but there is a difference. Perhaps this is causing the CI errors.

File: nltk/tokenize/punkt.py [hunk 2]
```diff
@@ -1321,8 +1321,8 @@ def span_tokenize(self, text, realign_boundaries=True):
         slices = self._slices_from_text(text)
         if realign_boundaries:
             slices = self._realign_boundaries(text, slices)
-        for sl in slices:
-            yield (sl.start, sl.stop)
+        for sentence in slices:
+            yield (sentence.start, sentence.stop)
```
Comment (tomaarsen) [inline nltk/tokenize/punkt.py:1325]: You use `sentence` all around, but I think these are meant to be slices? Like, `slice` objects that correspond to a sentence. `sentence` sounds like a string or a list of strings. Also, be careful, `sentence` is used as a list of tokens elsewhere in the code: 
https://github.com/nltk/nltk/blob/a31de6c687165afe64a490053c73c8e0b20ff6db/nltk/tokenize/punkt.py#L1402-L1415

File: nltk/tokenize/punkt.py [hunk 3]
```diff
@@ -1362,31 +1362,31 @@ def _realign_boundaries(self, text, slices):
...
+                if text[sentence1]:
+                    yield sentence1
 
     def text_contains_sentbreak(self, text):
         """
         Returns True if the given text includes a sentence break.
         """
         found = False  # used to ignore last token
-        for t in self._annotate_tokens(self._tokenize_words(text)):
+        for text in self._annotate_tokens(self._tokenize_words(text)):
```
Comment (stevenbird) [inline nltk/tokenize/punkt.py]: Prefer not to see `text` denoting a tuple and a string in the same line of code

File: nltk/tokenize/punkt.py [hunk 4]
```diff
@@ -570,8 +570,8 @@ def _tokenize_words(self, plaintext):
                 yield self._Token(tok, parastart=parastart, linestart=True)
                 parastart = False
 
-                for t in line_toks:
-                    yield self._Token(t)
+                for token in line_toks:
```
Comment (stevenbird) [inline nltk/tokenize/punkt.py]: Prefer to be consistent with abbreviations, e.g. `for tok in line_toks` or `for token in line_tokens`

**Comment by stevenbird** [comment]
(FYI @jnothman)

**Comment by jnothman** [comment]
> (FYI @jnothman)

No issues on my side. Good to see the code receiving some TLC

**Comment by stevenbird** [comment]
Thanks @12mohaned

**Pull request merged**